yarn-project/simulator/src/public/public_processor/public_processor.ts

```diff
@@ -87,8 +87,8 @@ export class PublicProcessorFactory {
     doMerkleOperations: boolean,
     skipFeeEnforcement: boolean,
     telemetryClient: TelemetryClient,
-  ) {
-    return new PublicTxSimulator(
+  ): TelemetryPublicTxSimulator {
```

Comment:
Should the caller care that it's a `TelemetryPublicTxSimulator`? since you are extending maybe just `PublicTxSimulator` would be enough? (ideally an interface but it's probably overkill)